packages/hardhat-viem/test/clients.ts

```diff
@@ -0,0 +1,165 @@
+import type { EthereumProvider } from "hardhat/types";
+
+import { assert } from "chai";
+import * as chains from "viem/chains";
+
+import {
+  innerGetPublicClient,
+  innerGetWalletClients,
+  innerGetTestClient,
+} from "../src/internal/clients";
+import { EthereumMockedProvider } from "./mocks/provider";
+
+describe("clients", () => {
+  describe("innerGetPublicClient", () => {
+    it("should return a public client", async () => {
+      const provider: EthereumProvider = new EthereumMockedProvider();
+
+      const client = await innerGetPublicClient(provider, chains.mainnet);
+
+      assert.isDefined(client);
+      assert.equal(client.type, "publicClient");
+      assert.equal(client.chain?.id, chains.mainnet.id);
```

Comment:
Unnecessary?

```suggestion
      assert.equal(client.chain.id, chains.mainnet.id);
```

There are other places where this is happening too. You can add `/* eslint @typescript-eslint/no-unnecessary-condition: "error" */` at the beginning of the file to find them.